### PR TITLE
Inventory page

### PR DESCRIPTION
**Pull request by miguelstorani**
Major changes in inventory component.

image

Inventory details modal
image

Tags:
image
image
image
image

Assign tags to multiple devices
image

### CODE REVIEW

File: src/components/inventory/inventory-utils/devices-affected-table.component.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            <Table.HeaderCell>NAME</Table.HeaderCell>
+            <Table.HeaderCell>VENDOR</Table.HeaderCell>
+          </Table.Row>
+        </Table.Header>
+          <Table.Body className="affected-devices-body-container">
+            {props.assets.filter((i) => i.selected).map((item, index) => 
+              <Table.Row key={index}>
+                <Table.Cell>{item.hex_id}</Table.Cell>
+                <Table.Cell>{item.name}</Table.Cell>
+                <Table.Cell>{item.vendor? item.vendor : "asfgzdfh "}</Table.Cell>
```
Comment (miguelstorani) [inline src/components/inventory/inventory-utils/devices-affected-table.component.js:27]: delete "asfgzdfh"

File: src/components/inventory/inventory.assign-tags.modal.component.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+const AssignTagsModal = (props) => {
+
+  /*
+    props:
+      open: boolean
+      assets: list of all asset on the page (with selected parameter)
+      onClose: function to run when closing modal without assigning tags
+      onSuccess: function to call when assig
```
Comment (miguelstorani) [inline src/components/inventory/inventory.assign-tags.modal.component.js:19]: delete line

File: src/components/inventory/inventory.assign-tags.modal.component.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      This component will assign the tags to all selected devices by itself,
+      there is no neeed to handle the close event, but it will triggger.
```
Comment (miguelstorani) [inline src/components/inventory/inventory.assign-tags.modal.component.js:22]: complete comment

File: src/components/inventory/inventory.assign-tags.modal.component.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  */
+ const { tagsStore } = React.useContext(MobXProviderContext);
+ const [open, setOpen] = React.useState(!!props.open);
+ const [tagsToAssign, setTagsToAssign] = React.useState([]);
+ const [assigning, setAssigning] = React.useState(false);
+ const [sendDisabled, setSendDisabled] = React.useState(true);
+
+
+  const ShowTags = (props) => {
+    console.log(tagsToAssign);
```
Comment (miguelstorani) [inline src/components/inventory/inventory.assign-tags.modal.component.js:32]: delete log

File: src/components/inventory/inventory.assign-tags.modal.component.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    Promise.all(tagsAssignPromise).then(() => {
+      setAssigning(false)
+      if (props.onClose instanceof Function) props.onClose();
+    }
+    );
+  }
+
+
+  console.log(tagsToAssign);
```
Comment (miguelstorani) [inline src/components/inventory/inventory.assign-tags.modal.component.js:58]: delete line

File: src/components/inventory/inventory.component.js [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+//import Bubble from "../visualizations/Bubble";
+import InventoryDetailsModal from "./inventory.modal.component";
+import AssignTagsModal from "./inventory.assign-tags.modal.component";
+
+import "./inventory.component.css";
+import LoaderComponent from "../utils/loader.component";
+import EmptyComponent from "../utils/empty.component";
+import InventoryIdComponent from "./inventory-id.component";
+import { withRouter } from "react-router-dom";
+import { selectAll } from "d3";
```
Comment (miguelstorani) [inline src/components/inventory/inventory.component.js:18]: delete unused imports

File: src/components/inventory/inventory.component.js [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+
+    this.state = {
+      isLoading: true,
+      isGraphsLoading: true,
+      activePage: 1,
+      pageSize: 50,
+      pagesCount: null,
+      assets: [],
+      assetsCount: null,
+			byVendorsViz: [],
```
Comment (miguelstorani) [inline src/components/inventory/inventory.component.js]: delete indentation

File: src/components/inventory/inventory.component.js [hunk 8]
```diff
@@ -0,0 +1,543 @@
...
+    const assetsPromise = this.props.inventoryAssetsStore.getAssets({page: activePage, size: pageSize}, criteria);
+    const dataCollectorsPromise = this.props.inventoryAssetsStore.getDataCollectorsCount(criteria);
+    const gatewaysPromise = this.props.inventoryAssetsStore.getGatewaysCount(criteria);
+    const vendorsPromise = this.props.inventoryAssetsStore.getVendorsCount(criteria);
+    const tagsPromise = this.props.inventoryAssetsStore.getTagsCount(criteria);
+
+    
+    Promise.all([assetsPromise, dataCollectorsPromise, gatewaysPromise, vendorsPromise, tagsPromise]).then(
+      (responses) => {
+        console.log(responses)
```
Comment (miguelstorani) [inline src/components/inventory/inventory.component.js]: delete line

File: src/components/inventory/inventory.component.js [hunk 9]
```diff
@@ -0,0 +1,543 @@
...
+
+        {/* set selected field from criteria */}
+
+        dataCollectorsPieData.forEach((item) => item.selected = criteria.dataCollectors.includes(item.code));
+        gatewaysPieData.forEach((item) => item.selected = criteria.gateways.includes(item.code));
+        vendorsPieData.forEach((item) => item.selected = criteria.vendors.includes(item.code));
+
+        assetsList.forEach((item) => item.selected = false);
+
+        console.log(tags);
```
Comment (miguelstorani) [inline src/components/inventory/inventory.component.js:111]: delete line

File: src/components/inventory/inventory.component.js [hunk 10]
```diff
@@ -0,0 +1,543 @@
...
+                      </Grid>
+                    )}
+                  </Segment>
+
+                  {selectedAsset && <InventoryDetailsModal loading={this.state.isLoading} selectedItem={selectedAsset} assets={this.state.assets} onClose={this.closeInventoryDetails} onNavigate={this.goToAlert}/>}
+                  {assignTags && <AssignTagsModal open={assignTags} assets={assets} onClose={() => {this.setState({assignTags: false}); window.location.reload(false);}}/>}
+                </div>
+              </div>
+            </div>
+					</div>
```
Comment (miguelstorani) [inline src/components/inventory/inventory.component.js]: delete indentation

File: src/components/utils/tags/tag.creator.component.js [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+										textColor="white"
+										fontSize="30px"
+										opacity={validTagName? "1" : "0.5"}
+									/>
+								</div>
+								<Input disabled={sended} nameClass="" placeholder="tag name" onChange={handleInput}/>
+								<div style={{textAlign:"center", margin: "15px"}}>
+										{colors.map((col, index) => <ColorPicker disabled={sended} selected={color} color={col} onColorSelect={handleColorSelect}/>)}
+								</div>
+							</div>
```
Comment (miguelstorani) [inline src/components/utils/tags/tag.creator.component.js:92]: delete indentation

File: src/components/visualizations/Pie.js [hunk 12]
```diff
@@ -194,7 +194,7 @@ class Pie extends Component {
           .duration(200)
           .style("opacity", 1);
 
-        let tooltipContent = `<div><strong>${d.data.label}</strong> (${d.data.value})</div>`;
+        let tooltipContent = `<div><strong>${d.data.label? d.data.label : "unknown"}</strong> (${d.data.label? d.data.label : "unknown"}})</div>`;
```
Comment (miguelstorani) [inline src/components/visualizations/Pie.js]: delete last curly brace `${d.data.label? d.data.label : "unknown"}}`

File: src/util/api.js [hunk 13]
```diff
@@ -50,6 +50,10 @@ const API = {
         throw error;
       }
     );
+  },
+
+  getHost: () => {
+    return API_HOST;
```
Comment (miguelstorani) [inline src/util/api.js:56]: delete changes

File: src/components/inventory/inventory.set-importance.modal.component.js [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+ const { tagsStore } = React.useContext(MobXProviderContext);
+ const [open, setOpen] = React.useState(!!props.open);
+ const [tagsToAssign, setTagsToAssign] = React.useState([]);
+ const [assigning, setAssigning] = React.useState(false);
+ const [sendDisabled, setSendDisabled] = React.useState(true);
+
+
+  const ShowTags = (props) => {
+    console.log(tagsToAssign);
```
Comment (cpfarher) [inline src/components/inventory/inventory.set-importance.modal.component.js:28]: delete this line...

**Pull request merged**